app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaWidget.kt

```diff
@@ -0,0 +1,65 @@
+package de.tum.`in`.tumcampusapp.component.ui.cafeteria.widget
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.widget.RemoteViews
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.activity.CafeteriaActivity
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.controller.CafeteriaManager
+import de.tum.`in`.tumcampusapp.component.ui.cafeteria.repository.CafeteriaLocalRepository
+import de.tum.`in`.tumcampusapp.database.TcaDb
+import de.tum.`in`.tumcampusapp.service.MensaWidgetService
+import de.tum.`in`.tumcampusapp.utils.Const
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+
+/**
+ * Implementation of Mensa Widget functionality.
+ * The Update intervals is set to 10 hours in mensa_widget_info.xml
+ */
+class MensaWidget : AppWidgetProvider() {
+
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+
+        val localRepository = CafeteriaLocalRepository(TcaDb.getInstance(context))
+        val mensaManager = CafeteriaManager(context)
+
+        val cafeteriaId = mensaManager.bestMatchMensaId
+        val cafeteria = localRepository.getCafeteria(cafeteriaId)
+
+        for (appWidgetId in appWidgetIds) {
+            val remoteViews = RemoteViews(context.packageName, R.layout.mensa_widget)
+
+            // Set the header for the Widget layout
+            cafeteria?.let {
+                remoteViews.setTextViewText(R.id.mensa_widget_header, it.name)
+            }
+
+            // Set the properly formatted date in the subhead
```

Comment:
This shows the current date, but it should show the date of the next menu. For instance, on Saturday, it should show the menu for Monday.